tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go

```diff
@@ -228,6 +229,96 @@ func (r *RKE2NodeDriverProvisioningTestSuite) ProvisioningRKE2ClusterDynamicInpu
 	}
 }
 
+func (r *RKE2NodeDriverProvisioningTestSuite) ProvisioningRKE2CNICluster(provider Provider) {
+	providerName := " Node Provider: " + provider.Name
+	r.cnis = []string{"canal", "cilium", "calico", "multus,canal", "multus,cilium", "multus,calico"}
+
+	nodeRoles1 := []machinepools.NodeRoles{
+		{
+			ControlPlane: true,
+			Etcd:         false,
+			Worker:       false,
+			Quantity:     2,
+		},
+		{
+			ControlPlane: false,
+			Etcd:         true,
+			Worker:       false,
+			Quantity:     3,
+		},
+		{
+			ControlPlane: false,
+			Etcd:         false,
+			Worker:       true,
+			Quantity:     3,
+		},
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles []machinepools.NodeRoles
+		client    *rancher.Client
+	}{
+		{"3 etcd 2 cp 3 worker nodes Admin User", nodeRoles1, r.client},
+		{"3 etcd 2 cp 3 worker nodes Standard User", nodeRoles1, r.standardUserClient},
+	}
+
+	var name string
+	for _, tt := range tests {
+		subSession := r.session.NewSession()
+		defer subSession.Cleanup()
+
+		client, err := tt.client.WithSession(subSession)
+		require.NoError(r.T(), err)
+
+		cloudCredential, err := provider.CloudCredFunc(client)
+		require.NoError(r.T(), err)
+		for _, kubeVersion := range r.kubernetesVersions {
+			name = tt.name + providerName + " Kubernetes version: " + kubeVersion
+			for _, cni := range r.cnis {
```

Comment:
This line is taken from the config file in this format in the provisioning input like this 
```
` "provisioningInput": {
    "kubernetesVersion": ["v1.23.7+rke2r2"],
    "cni": ["calico", "canal"],`
}
```